### DISCUSSION TITLE
Uninstall?

### DISCUSSION THREAD

**Discussion by mtnsec** [Security-Onion-Solutions/securityonion#2331] [Unsupported Versions] [answered]
I used the network installer on Ubuntu 18.04 to set up a manager/search system and a separate sensor system.

I just rebuilt the manager from scratch, including the OS because I couldn't find a way to do a clean uninstall of security onion. Now, re-running setup on my sensor to connect to the "new" manager is causing all sorts of salt issues (fixing one, just leads to another, etc.), so I'm wondering if I'm going to have to do the same thing -- reload the OS and security onion - on the sensor as well.

For future reference, is there a way to cleanly uninstall security onion without having reinstall the entire OS?

Thanks!

**Comment by weslambert** [comment]
> I used the network installer on Ubuntu 18.04 to set up a manager/search system and a separate sensor system.
> 
> I just rebuilt the manager from scratch, including the OS because I couldn't find a way to do a clean uninstall of security onion. Now, re-running setup on my sensor to connect to the "new" manager is causing all sorts of salt issues (fixing one, just leads to another, etc.), so I'm wondering if I'm going to have to do the same thing -- reload the OS and security onion - on the sensor as well.
> 
> For future reference, is there a way to cleanly uninstall security onion without having reinstall the entire OS?
> 
> Thanks!

We are currently working on improving the reinstall process (ex. re-running `sosetup`) so that you don't need to perform a re-install of the OS when you run into issues like this.

**Reply by wompRS** [reply]
Has there been any update on this? Is there any functionality for uninstalling a sensor or some such and overwriting it? Currently I changed my manager name and cannot get the sensor overwritten as it gets stuck permanently on "configuring startup etc etc"

**Comment by kmfreder1** [comment]
I am wondering the same.  How do you cleanly uninstall a sensor and then reinstall.

**Comment by dougburks** [comment, marked as answer]
Re-running Setup should be improved in the upcoming 2.3.90 release:
https://github.com/Security-Onion-Solutions/securityonion/issues/4995
https://github.com/Security-Onion-Solutions/securityonion/issues/5588

**Reply by kmfreder1** [reply]
That is great news!  Thank you!

**Reply by BishalLakha** [reply]
Since 2.3.91 is already out, is there documentation for this ? Thank you

**Reply by dougburks** [reply]
You can re-run Setup by running the same `so-setup` or `so-setup-network` that you did originally:
https://docs.securityonion.net/en/2.3/installation.html

We have some additional improvements for this in the upcoming 2.3.100 release.
